exp/api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -94,6 +95,14 @@ func (r *AzureManagedMachinePool) ValidateUpdate(oldRaw runtime.Object, client c
 		}
 	}
 
+	if !reflect.DeepEqual(r.Spec.Taints, old.Spec.Taints) {
```

Comment:
how come taints are immutable? are they immutable in AKS pools?

Reply:
I took it as immutable from
```
--node-taints
The node taints for the node pool. You can't change the node taints through CLI after the node pool is created
```
https://docs.microsoft.com/en-us/cli/azure/aks/nodepool?view=azure-cli-latest#az-aks-nodepool-upgrade

Reply:
I will do a test for this upgrade

Reply:
they are immutable still. But while checking taints I discovered labels aren't immutable anymore.
In my last PR I couldn't update those. 
```
Label can only be set for node pools during node pool creation. Labels must also be a key/value pair and have a valid syntax.
```
![](https://i.imgur.com/LfOHUhQ.png)
now its gone from the documentation also...

Reply:
ok, the node labels I just forgot that I saw the doc change myself also and didn't add any webhook validation for them being immutable.
you can ignore my last comment.